exec/cnex/cnex.c

```diff
@@ -28,28 +29,82 @@ typedef struct tagTCommandLineOptions {
     int  ArgStart;
     char *pszFilename;
     char *pszExecutableName;
+    char *pszExecutablePath;
 } TOptions;
 
-static TOptions gOptions = { FALSE, FALSE, TRUE, 0, NULL, NULL };
+static TOptions gOptions = { FALSE, FALSE, TRUE, 0, NULL, NULL, NULL };
 
-void exec_freeExecutor(TExecutor *e)
+static char *getApplicationName(char *arg)
 {
-    uint32_t i;
+    char *p = arg;
+    char *s = p;
+
+    while (*p) {
+        if (*p == '\\' || *p == '/') {
+            s = p;
+        }
+        p++;
+    }
+    return ++s;
+}
 
-    for (i = 0; i < e->object->global_size; i++) {
-        cell_clearCell(&e->globals[i]);
+static char *getApplicationPath(char *arg)
+{
+    BOOL bHavePath = TRUE;
+    char *p = arg;
+    char *e = p;
+
+    while (*p) {
+        if (*p == '\\' || *p == '/') {
+            e = p;
+        }
+        p++;
+    }
+
+    if (e == p) {
```

Comment:
I think this needs some more careful handling here. If `e == p`, what does that mean exactly? I can only see that happening when `*arg == 0`. Why is `e` being decremented by 2? That would put it before `arg`. Then on line 69, you would be allocating 0 bytes, and trying to store things there. Finally, if `e < arg` then `r[e - arg]` on line 76 definitely does not look right.